File: typescript/resistor-color-duo/resistor-color-duo.ts
export class ResistorColor {
  private static mapping: string[] = [
    "black",
    "brown",
    "red",
    "orange",
    "yellow",
    "green",
    "blue",
    "violet",
    "grey",
    "white",
  ];

  private colors: string[];

  constructor(colors: string[]) {
    if (colors.length < 2) {
      throw "At least two colors need to be present";
    }
    this.colors = colors.slice(0, 2);
  }

  value = () =>
    Number.parseInt(
      this.colors.map((v) => ResistorColor.mapping.indexOf(v)).join("")
    );
}

File: ruby/resistor-color-duo/resistor_color_duo.rb
module ResistorColorDuo
  TRANSLATE = {
    'black' => '0',
    'brown' => '1',
    'red' => '2',
    'orange' => '3',
    'yellow' => '4',
    'green' => '5',
    'blue' => '6',
    'violet' => '7',
    'grey' => '8',
    'white' => '9'
  }.freeze

  def self.value(input)
    input.take(2).map(&TRANSLATE).join('').to_i
  end
end
